src/librustc/session/config.rs

```diff
@@ -1195,6 +1199,24 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         early_error(error_format, "Value for codegen units must be a positive nonzero integer");
     }
 
+    let mut prints = Vec::<PrintRequest>::new();
+    if cg.target_cpu.as_ref().map_or(false, |s| s == "help") {
+        prints.push(PrintRequest::TargetCPUs);
+        cg.target_cpu = None;
```

Comment:
Push as print request and reset target-cpu vaule so LLVM doesn't print help several times when creating a target machine, we want to print ourselves from rustc.